### PR TITLE
ID-1493 [Feat] New operator for executing LFD filters

### PR DESCRIPTION
**Pull request by tonytlwu**
Ref. https://weboo.atlassian.net/browse/ID-1493

### CODE REVIEW

File: sift.js [hunk 1]
```diff
@@ -81,6 +98,350 @@
     return true;
   }
 
+  function smartParseFloat(value) {
```
Comment (squallstar) [inline sift.js:106]: Add JS doc for this function

File: sift.js [hunk 2]
```diff
@@ -81,6 +98,350 @@
...
+        return value;
+      }
+
+      return ('' + value).trim();
+    }).filter(function(value) {
+      return Array.isArray(value) || [undefined, null, '', NaN].indexOf(value) === -1;
+    });
+  }
+
+  function getCachedDate(offsetType, offsetValue, getDate) {
```
Comment (squallstar) [inline sift.js]: Add JS doc for this function

File: sift.js [hunk 3]
```diff
@@ -81,6 +98,350 @@
...
+    } else if (offsetType.indexOf('today') !== -1) {
+      currentDate[offsetType] = offsetType.indexOf('add') !== -1
+        ? moment().add(period, smartParseFloat(offsetValue)).startOf('day')
+        : moment().subtract(period, smartParseFloat(offsetValue)).startOf('day');
+    }
+
+    return currentDate[offsetType];
+  }
+
+  function getInputDate(date, getDate, timeOnly, dateOnly) {
```
Comment (squallstar) [inline sift.js]: Add JS doc for this function

File: sift.js [hunk 4]
```diff
@@ -81,6 +98,350 @@
...
+    if (!dateOnly) {
+      inputDate = timeOnly
+        ? getDate(date, 'HH:mm')
+        : getDate(date, DATE_FORMAT);
+    }
+
+    return inputDate;
+  }
+
+  function getDateModifiedValues(options) {
```
Comment (squallstar) [inline sift.js]: Add JS doc for this function

File: sift.js [hunk 5]
```diff
@@ -81,6 +98,350 @@
...
+            ? null
+            : moment(options.date, DATE_FORMAT).startOf('day')
+        };
+
+      default:
+        break;
+    }
+  }
+
+  function isDateMatches(options) {
```
Comment (squallstar) [inline sift.js]: Add JS doc for this function

**Review by squallstar — COMMENTED** [review]
Great work! I left you some comments to add JSDoc for all functions you added.

File: sift.js (hunk 1 above)
Comment (tonytlwu) [reply sift.js:106]: Added

File: sift.js (hunk 2 above)
Comment (tonytlwu) [reply sift.js]: As [this comment](https://weboo.atlassian.net/browse/ID-1493?focusedCommentId=25714) suggested, the PR will remain open. https://weboo.atlassian.net/browse/ID-1802 will require some of these date comparison functions and helper functions to be rewritten. Some of them may become redundant.

Once they're all cleaned up, I'll add the JSDoc for you to review.

**Comment by tonytlwu** [comment]
@squallstar Date filter conditions have now also been added. Please review.

Ref. https://weboo.atlassian.net/browse/ID-1802

**Review by squallstar — DISMISSED** [review]

**Review by squallstar — DISMISSED** [review]

**Pull request merged**